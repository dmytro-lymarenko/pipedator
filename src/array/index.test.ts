import { array } from './index';

describe('array()', () => {
	it('should validate an array', () => {
		expect(array().validate([])).toBeNull();
	});

	it('should return an error for non-array values', () => {
		expect(array().validate(5)).toMatchSnapshot();
		expect(array().validate({})).toMatchSnapshot();
		expect(array().validate('array')).toMatchSnapshot();
		expect(array().validate(() => {})).toMatchSnapshot();
		expect(array().validate(null)).toMatchSnapshot();
		expect(array().validate(undefined)).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(array('Custom message').validate(5)).toMatchSnapshot();
	});
});
