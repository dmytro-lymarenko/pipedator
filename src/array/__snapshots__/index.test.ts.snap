// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array() should return an error for non-array values 1`] = `
Object {
  "children": null,
  "message": "Value should be an array",
  "path": "",
}
`;

exports[`array() should return an error for non-array values 2`] = `
Object {
  "children": null,
  "message": "Value should be an array",
  "path": "",
}
`;

exports[`array() should return an error for non-array values 3`] = `
Object {
  "children": null,
  "message": "Value should be an array",
  "path": "",
}
`;

exports[`array() should return an error for non-array values 4`] = `
Object {
  "children": null,
  "message": "Value should be an array",
  "path": "",
}
`;

exports[`array() should return an error for non-array values 5`] = `
Object {
  "children": null,
  "message": "Value should be an array",
  "path": "",
}
`;

exports[`array() should return an error for non-array values 6`] = `
Object {
  "children": null,
  "message": "Value should be an array",
  "path": "",
}
`;

exports[`array() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
